client/tests/integration/events/pipeline.rs

```diff
@@ -15,98 +11,68 @@ use super::Configuration;
 const PEER_COUNT: usize = 7;
 
 #[test]
-fn transaction_event_should_be_sent_to_all_peers_from_all_peers() -> Result<()> {
-    test_with_instruction_and_status(None, true)?;
-    let fail = FailBox::new("Failing transaction to test Rejected event.");
-    test_with_instruction_and_status(Some(fail.into()), false)
+fn transaction_with_no_instructions_should_be_committed() -> Result<()> {
+    test_with_instruction_and_status(None, PipelineStatusKind::Committed)
+}
+
+#[test]
+fn transaction_with_fail_instruction_should_be_rejected() -> Result<()> {
+    let fail = FailBox::new("Should be rejected");
+    test_with_instruction_and_status(Some(fail.into()), PipelineStatusKind::Rejected)
 }
 
 #[allow(clippy::needless_range_loop, clippy::needless_pass_by_value)]
 fn test_with_instruction_and_status(
     instruction: Option<Instruction>,
-    should_be_committed: bool,
+    should_be: PipelineStatusKind,
 ) -> Result<()> {
-    let (_rt, network, _) = <Network>::start_test_with_runtime(PEER_COUNT.try_into().unwrap(), 1);
-    wait_for_genesis_committed(&network.clients(), 0);
+    let (_rt, network, genesis_client) =
+        <Network>::start_test_with_runtime(PEER_COUNT.try_into().unwrap(), 1);
+    let clients = network.clients();
+    wait_for_genesis_committed(&clients, 0);
+    let pipeline_time = Configuration::pipeline_time();
+    // Given
+    let submitter = genesis_client;
+    let transaction = submitter.build_transaction(instruction.into(), UnlimitedMetadata::new())?;
+    let hash = transaction.hash();
+    let mut handles = Vec::new();
+    for listener in clients {
+        let checker = Checker { listener, hash };
+        let handle_validating = checker.clone().spawn(PipelineStatusKind::Validating);
+        handles.push(handle_validating);
+        let handle_validated = checker.spawn(should_be);
+        handles.push(handle_validated);
+    }
+    // When
+    submitter.submit_transaction(transaction)?;
+    thread::sleep(pipeline_time * 2);
+    // Then
+    for handle in handles {
+        handle.join().expect("Thread panicked")
+    }
+    Ok(())
+}
 
-    for submitting_peer in 0..PEER_COUNT {
```

Comment:
I don't think n:n transactions should be tested. 1:n should be enough

Reply:
We can always add n:n tests back in. With that said, I'd prefer adding a property-based suite.